src/Lachain.Core/Blockchain/VM/InvocationContext.cs

```diff
@@ -11,7 +12,7 @@ public class InvocationContext
 
         public UInt256 Value => Receipt.Transaction.Value;
 
-        public UInt256 TransactionHash => Receipt.FullHash();
+        public UInt256 TransactionHash => Receipt.FullHash(HardforkHeights.IsHardfork_9Active(Snapshot.Blocks.GetTotalBlockHeight() + 1));
```

Comment:
I think it could be HardforkHeights.IsHardfork_9Active(Receipt.Block)

Reply:
If the Receipt.Block is not null

Reply:
Why do you think so?

Reply:
Sorry, my bad. I think Receipt.Block could be null.